tests/perf-system/benchmarks/benchmarks.py

```diff
@@ -0,0 +1,513 @@
+        commit_tx = json.loads(commit_tx_process.stdout.decode())["transaction_id"]
+        assert int(commit_tx.split(".")[0]) == 2, f"Commit tx is {commit_tx}"
+
+        LOG.info("Checking message count is zero before experiment")
+        count_process = subprocess.run(
+            [
+                "curl",
+                address + "/log/public/count",
+                *user_cert_args,
+            ],
+            capture_output=True,
+        )
+        message_count_before = int(count_process.stdout.decode())
+        assert (
+            message_count_before == 0
+        ), f"Message count before test is {message_count_before}"
+
+        LOG.info("Checking consensus state")
+        check_consensus_process = subprocess.run(
+            [
+                "curl",
+                address + "/node/consensus",
+                *user_cert_args,
+            ],
+            capture_output=True,
+        )
+
+        consensus_state = json.loads(check_consensus_process.stdout.decode())
+        node_leadership_state = consensus_state["details"]["leadership_state"]
+        if id == 0:
+            assert node_leadership_state == "Leader", f"Node {id} is not leader"
+        else:
+            assert node_leadership_state == "Follower", f"Node {id} is not follower"
+
+        LOG.info(f"Finished checking node {id} at {address} is ready")
+        return ccf_server_process
+
+
+def shutdown_experiment(
+    ccf_server_process,
+    ccf_node_ips,
+    user_cert_args,
+    number_of_requests,
+    ccf_node_for_piccolo,
+    write_percentages,
+    experiment_dir,
+    ccf_dir,
+):
+    LOG.info("Confirming CCF service state after experiments")
+
+    addresses_of_ccf_nodes = [f"https://{i}" for i in ccf_node_ips]
+    for id, address in enumerate(addresses_of_ccf_nodes):
+        LOG.info(f"Checking node {id} at {address}")
+
+        LOG.info("Checking message count")
+        message_count_process = subprocess.run(
+            [
+                "curl",
+                address + "/log/public/count",
+                *user_cert_args,
+            ],
+            capture_output=True,
+        )
+        message_count_after = int(message_count_process.stdout)
+        assert (
+            number_of_requests == message_count_after
+        ), f"Expected {number_of_requests} messages, got {message_count_after}"
+
+        LOG.info("Checking API metrics")
+
+        requests_per_workload = number_of_requests * ccf_node_for_piccolo.count(
+            ccf_node_ips[id]
+        )
+        total_requests = requests_per_workload * len(write_percentages)
+        total_posts = 0
+        [
+            total_posts := total_posts + requests_per_workload * w / 100
+            for w in write_percentages
+        ]
+        total_gets = total_requests - total_posts
+        LOG.info(
+            f"Expecting {total_requests} requests, comprised of {total_posts} posts and {total_gets} gets"
+        )
+
+        metrics_process = subprocess.run(
```

Comment:
Using https://github.com/microsoft/CCF/blob/9e0e46c41a306d10dc128f3f3672ad16a58cc089/tests/infra/clients.py#L544 will be shorter, provide good logging, and will run quicker since you will need fewer tls handshakes and file operations.